pkg/network/setup/podnic.go

```diff
@@ -126,7 +126,13 @@ func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netd
 }
 
 func (l *podNIC) setPodInterfaceCache() error {
-	ifCache := &cache.PodIfaceCacheData{Iface: l.vmiSpecIface}
+	ifCache, err := cache.ReadPodInterfaceCache(l.cacheCreator, string(l.vmi.UID), l.vmiSpecNetwork.Name)
+	if err != nil {
```

Comment:
Please differentiate between NotExist error and any other error. On any other error please return err, otherwise existing data in the file may be overridden.

Reply:
Done